storage: TestMVCCStatsPutWaitDeleteGC failed

storage.TestMVCCStatsPutWaitDeleteGC [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13204464?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/13204464?buildTab=artifacts#/) on release-23.2 @ [6cc2a2574a6877ea3eb674050e05373ebee6ccc3](https://github.com/cockroachdb/cockroach/commits/6cc2a2574a6877ea3eb674050e05373ebee6ccc3):

Fatal error:

```
panic: test timed out after 4m55s
running tests:
	TestMVCCStatsPutWaitDeleteGC (0s)

```
Stack: 

```
goroutine 18323 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x320
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x38
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestMVCCStatsPutWaitDeleteGC
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/df49cd4529be694977d51f20f1abedea/logTestMVCCStatsPutWaitDeleteGC2173614302
    test_log_scope.go:81: use -show-logs to present logs inline
    mvcc_stats_test.go:1229: -- test log scope end --
```
</p>
</details>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/storage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMVCCStatsPutWaitDeleteGC.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-34724

--- a/pkg/storage/BUILD.bazel
+++ b/pkg/storage/BUILD.bazel
@@ -142,7 +142,7 @@ go_test(
     data = glob(["testdata/**"]),
     embed = [":storage"],
     exec_properties = {"Pool": "large"},
-    shard_count = 2,
+    shard_count = 4,
     deps = [
         "//pkg/base",
         "//pkg/clusterversion",

